includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2256,11 +2272,30 @@ protected function get_advanced_fraud_protection_settings() {
 			return [];
 		}
 
-		$settings = get_option( 'advanced_fraud_protection_settings', false );
-		if ( ! $settings ) {
-			return Fraud_Risk_Tools::get_default_protection_settings();
+		// Check if we have local cache available before pulling it from the server.
+		$cached_server_settings = get_transient( 'wcpay_fraud_protection_settings' );
```

Comment:
This is the place where all magic happens. If there's no ruleset on the server saved for  the merchant, we save the "Basic" protection level ruleset as the first one automatically. This part needs a deep investigation.